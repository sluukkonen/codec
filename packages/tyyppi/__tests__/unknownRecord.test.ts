import { unknownRecord } from "../src/index.js"
import { expectParseFailure, expectParseSuccess } from "./helpers.js"

test("should parse any object", () => {
  expectParseSuccess(unknownRecord, {})
  expectParseSuccess(unknownRecord, { a: 1 })
})

test("should not parse null", () => {
  expectParseFailure(unknownRecord, null)
})

test("should not parse an array", () => {
  expectParseFailure(unknownRecord, [])
})
