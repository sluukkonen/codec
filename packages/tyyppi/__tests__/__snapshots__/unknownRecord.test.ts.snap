// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should not parse an array 1`] = `
[
  {
    "code": "invalid_type",
    "expected": "object",
    "message": "Expected an object, received an array",
    "path": [],
    "received": "array",
  },
]
`;

exports[`should not parse null 1`] = `
[
  {
    "code": "invalid_type",
    "expected": "object",
    "message": "Expected an object, received null",
    "path": [],
    "received": "null",
  },
]
`;
